portal-bridge/src/types/full_header.rs

```diff
@@ -77,6 +80,40 @@ impl TryFrom<Value> for FullHeader {
     }
 }
 
+impl FullHeader {
+    pub fn validate(&self) -> anyhow::Result<()> {
+        // validation for pre-merge blocks
+        if self.header.number < MERGE_BLOCK_NUMBER {
+            let epoch_acc = self
+                .epoch_acc
+                .as_ref()
+                .ok_or_else(|| anyhow!("epoch_acc is missing for pre-merge block"))?;
+
+            // Fetch HeaderRecord from EpochAccumulator for validation
+            let header_index = self.header.number % EPOCH_SIZE;
+            let header_record = &epoch_acc[header_index as usize];
+
+            // Validate Header
+            let actual_header_hash = self.header.hash();
+            if header_record.block_hash != actual_header_hash {
```

Comment:
nitpick: Can we leverage here and on line 106 [anyhow::ensure](https://docs.rs/anyhow/latest/anyhow/macro.ensure.html#)?